includes/User.php

```diff
@@ -2169,6 +2167,8 @@ public function invalidateCache() {
 					__METHOD__ );
 			}
 
+			$dbw->commit();
```

Comment:
The `selectField` statement above ends up starting a `BEGIN` statement on this connection. We need to explicitly `COMMIT` it here or the update above won't persist.